arangod/Aql/CollectionAccessingNode.cpp

```diff
@@ -152,16 +163,17 @@ aql::Collection const* CollectionAccessingNode::collection() const {
   return _collectionAccess.collection();
 }
 
-void CollectionAccessingNode::restrictToShard(std::string const& shardId) {
+void CollectionAccessingNode::restrictToShard(ShardID const& shardId) {
   _restrictedTo = shardId;
 }
 
 bool CollectionAccessingNode::isRestricted() const {
-  return !_restrictedTo.empty();
+  return _restrictedTo.has_value();
 }
 
-std::string const& CollectionAccessingNode::restrictedShard() const {
-  return _restrictedTo;
+ShardID const& CollectionAccessingNode::restrictedShard() const {
```

Comment:
As ShardIDs are now light-weight, we should now make this function return a copy of the ShardID.
```suggestion
ShardID CollectionAccessingNode::restrictedShard() const {
```